packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/diagramfilter/services/DiagramFilterDescriptionProvider.java

```diff
@@ -0,0 +1,333 @@
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.eclipse.sirius.components.collaborative.api.IEditingContextEventProcessor;
+import org.eclipse.sirius.components.collaborative.api.IEditingContextEventProcessorRegistry;
+import org.eclipse.sirius.components.collaborative.api.IRepresentationEventProcessor;
+import org.eclipse.sirius.components.collaborative.diagrams.api.DiagramImageConstants;
+import org.eclipse.sirius.components.collaborative.diagrams.api.IDiagramEventProcessor;
+import org.eclipse.sirius.components.collaborative.forms.api.IDiagramFilterDescriptionProvider;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.components.diagrams.CollapsingState;
+import org.eclipse.sirius.components.diagrams.Diagram;
+import org.eclipse.sirius.components.diagrams.Node;
+import org.eclipse.sirius.components.diagrams.ViewModifier;
+import org.eclipse.sirius.components.forms.WidgetIdProvider;
+import org.eclipse.sirius.components.forms.description.AbstractControlDescription;
+import org.eclipse.sirius.components.forms.description.CheckboxDescription;
+import org.eclipse.sirius.components.forms.description.FormDescription;
+import org.eclipse.sirius.components.forms.description.GroupDescription;
+import org.eclipse.sirius.components.forms.description.LabelDescription;
+import org.eclipse.sirius.components.forms.description.PageDescription;
+import org.eclipse.sirius.components.forms.description.SplitButtonDescription;
+import org.eclipse.sirius.components.forms.description.TreeDescription;
+import org.eclipse.sirius.components.representations.GetOrCreateRandomIdProvider;
+import org.eclipse.sirius.components.representations.Success;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.web.application.views.diagramfilter.services.api.IDiagramFilterActionContributionProvider;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+/**
+ * Provides a tree description listing diagram elements and allowing to perform actions on them.
+ *
+ * @author gdaniel
+ */
+@Service
+public class DiagramFilterDescriptionProvider implements IDiagramFilterDescriptionProvider {
+
+    public static final String FORM_DESCRIPTION_ID = "diagram_filter_form_description";
+
+    public static final String SELECTED_TREE_NODES = "selectedTreeNodes";
+
+    private static final String GROUP_DESCRIPTION_ID = "defaultDiagramFilterGroup";
+
+    private static final String PAGE_DESCRIPTION_ID = "defaultDiagramFilterPage";
+
+    private static final String FORM_TITLE = "Diagram Filters";
+
+    private final IObjectService objectService;
+
+    private IEditingContextEventProcessorRegistry editingContextEventProcessorRegistry;
+
+    private List<IDiagramFilterActionContributionProvider> splitButtonContributionProvider;
```

Comment:
Since you have multiple providers, the name of the field should reflect that

Reply:
done